### ISSUE TITLE
std::binary_function removed on c++17

### ISSUE THREAD

**Issue by imiskolee** [MTG/essentia#1206]
Hello, can not build from source using cpp 17 , beacuse of used removed class std::binary_function

**Comment by yyf** [comment]
@imiskolee I ran into the same issue. Any luck finding a workaround for this issue?

**Comment by imiskolee** [comment]
> @imiskolee I ran into the same issue. Any luck finding a workaround for this issue?

yes, just remove it, but you will face a lots another issues like them.
